cumulus/client/consensus/aura/src/collators/lookahead.rs

```diff
@@ -399,7 +399,16 @@ where
 					)
 					.await
 				{
-					Ok(Some((collation, block_data, new_block_hash))) => {
+					Ok(Some((collation, block_data))) => {
+						let Some(new_block_header) =
+							block_data.blocks().first().map(|b| b.header().clone())
+						else {
+							tracing::error!(target: crate::LOG_TARGET,  "Produced PoV doesn't contain any blocks");
+							break
+						};
+
+						let new_block_hash = new_block_header.hash();
+
 						// Here we are assuming that the import logic protects against equivocations
 						// and provides sybil-resistance, as it should.
 						collator.collator_service().announce_block(new_block_hash, None);
```

Comment:
I am having troubles to undestand why only first block's hash is announced? Shouldn't we announce hashes of all blocks?

Reply:
Lookahead is only producing one block.